arclight_installer.go

```diff
@@ -0,0 +1,195 @@
+package installer
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+type (
+	ArclightInstaller struct {
+	}
+
+	ArclightRelease struct {
+		Assets      []ArclightAssets `json:"assets"`
+		IsExpired   bool
+		PublishTime string `json:"published_at"`
+	}
+
+	ArclightAssets struct {
+		AssetsUrl   string `json:"url"`
+		AssetsName  string `json:"name"`
+		DownloadUrl string `json:"browser_download_url"`
+	}
+)
+
+var DefaultArclightInstaller = &ArclightInstaller{}
+
+var _ Installer = DefaultArclightInstaller
+
+func init() {
+	Installers["arclight"] = DefaultArclightInstaller
+}
+
+func (r *ArclightInstaller) Install(path, name string, target string) (installed string, err error) {
+	return r.InstallWithLoader(path, name, target, "")
+}
+
+func (r *ArclightInstaller) InstallWithLoader(path, name string, target string, loader string) (installed string, err error) {
+	data, err := r.GetInstallerVersions()
+	if err != nil {
+		return "", err
+	}
+	if len(loader) == 0 {
+		allVersions := r.GetOnlyVersions(data)
+		if target == "latest" {
+			loader, err = r.GetLatestVersion()
+			if err != nil {
+				return "", err
+			}
+			goto DownloadPart
+		}
+		for i := 0; i < len(allVersions); i += 1 {
+			if allVersions[i] == target {
+				loader = target
+				goto DownloadPart
+			}
+		}
+		loger.Info("not find the suitable builder, the version should be included in the following list:")
+		for i := 0; i < len(allVersions); i += 1 {
+			if data[allVersions[i]].IsExpired == true {
```

Comment:
为什么我还是能遇到 == true 这种操作 😓 

```suggestion
			if data[allVersions[i]].IsExpired {
```

Reply:
哦我明白了 false和nil有内存区别，c的判断逻辑是nil和false是否